### PR TITLE
Added error messages on signin and signup + UI tweaks

### PR DESCRIPTION
**Pull request by Jhoscar22**
## Describe your changes
Added error messages in /signup and /signin pages
Eliminated Name and Last-name fields in signup (unnecessary due to absence in backend)
Fixed signin form fields width to be consistent with other pages.

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

File: data-dashboard/pages/signin.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { IbmButton, TextInput, InputField } from '@/lib/components';
 import trpc from '@/lib/hooks/trpc';
 import TopLayout from '@/lib/components/TopLayout/TopLayout';
 import { NextPageWithLayout } from './_app';
+import { set } from 'zod';
```
Comment (antonieto) [inline data-dashboard/pages/signin.tsx:12]: pls remove this unused import
```suggestion
```

File: data-dashboard/pages/signin.tsx [hunk 2]
```diff
@@ -79,13 +80,24 @@ const ImageStyle = {
   marginRight: '202px',
 };
 
+const ErrorMessage = styled.p`
+  font-family: Verdana, Geneva, Tahoma, sans-serif;
```
Comment (antonieto) [inline data-dashboard/pages/signin.tsx:84]: Is it possible to move this into a global styles file?

File: data-dashboard/pages/signin.tsx [hunk 3]
```diff
@@ -79,13 +80,24 @@ const ImageStyle = {
...
+  font-size: 14px;
+  font-weight: normal;
+  color: #e71d36;
+  margin-top: 8px;
+`;
+
 function SignIn() {
   const router = useRouter();
+  const [error, setError] = useState(false);
```
Comment (antonieto) [inline data-dashboard/pages/signin.tsx:93]:
```suggestion
```

File: data-dashboard/pages/signin.tsx [hunk 4]
```diff
@@ -79,13 +80,24 @@ const ImageStyle = {
...
   const { mutate } = trpc.auth.login.useMutation({
```
Comment (antonieto) [inline data-dashboard/pages/signin.tsx:94]: Please use trpc error instead:
```suggestion
  const { mutate, error } = trpc.auth.login.useMutation({
```

File: data-dashboard/pages/signup.tsx [hunk 5]
```diff
@@ -90,18 +96,78 @@ function SignUp() {
...
     confirmPassword: '',
   });
 
+  const [error, setError] = useState({
+    email: '',
+    password: '',
+    confirmPassword: '',
+  });
+
+  const validatePassword = (password: string) => {
```
Comment (antonieto) [inline data-dashboard/pages/signup.tsx:110]: You might want to take a look at `z.string()` :)

File: data-dashboard/pages/signup.tsx [hunk 6]
```diff
@@ -90,18 +96,78 @@ function SignUp() {
   });
 
   const [credentials, setCredentials] = useState({
-    name: '',
     email: '',
     password: '',
     confirmPassword: '',
   });
 
+  const [error, setError] = useState({
```
Comment (antonieto) [inline data-dashboard/pages/signup.tsx:104]: Most of the time error states are unnecessary when we are using trpc. Could you please remove this and use the error returned by trpc?
```suggestion
```

File: data-dashboard/pages/signup.tsx (hunk 5 above)
Comment (antonieto) [inline data-dashboard/pages/signup.tsx:110]: By the way, it seems like these methods don't belong in this file and component. First of all, the component is loaded with too much unrelated logic, you are redefining this method every time this component is re-rendered, which reduces performance. Second, you are making this component really hard to read as it has too much unrelated logic. Consider moving this method outside the component into `@/lib/utils`.

**Review by antonieto — COMMENTED** [review]
good job!

**Review by DavBot02 — APPROVED** [review]

**Pull request merged**